packages/legend-studio/src/stores/editor-state/element-editor-state/RuntimeEditorState.ts

```diff
@@ -92,6 +91,27 @@ export const getClassMappingStore = (
   } else if (sourceElement instanceof TableAlias) {
     return sourceElement.relation.ownerReference.value;
   }
+  if (sourceElement) {
+    const extraInstanceSetImplementationStoreGetters = editorStore.pluginManager
+      .getStudioPlugins()
+      .flatMap(
+        (plugin) =>
+          (
+            plugin as DSLMapping_LegendStudioPlugin_Extension
+          ).getExtraInstanceSetImplementationStoreGetters?.() ?? [],
+      );
+    for (const instanceSetImplementationStoreGetter of extraInstanceSetImplementationStoreGetters) {
+      const instanceSetImplementationStore =
+        instanceSetImplementationStoreGetter(sourceElement);
+      if (instanceSetImplementationStore) {
+        return instanceSetImplementationStore;
+      }
+    }
+    throw new UnsupportedOperationError(
+      `Can't find suitable store for class mapping: no compatible store available from plugins`,
+      setImplementation,
+    );
+  }
   return undefined;
 };
```

Comment:
@akphi do you there would be an issue if I throw an error when i don't find any store instead of returning `undefined`?